python/lsst/resources/_resourcePath.py

```diff
@@ -485,22 +493,24 @@ def parent(self) -> ResourcePath:
         For a file-like URI this will be the same as calling `dirname()`.
         """
         # When self is file-like, return self.dirname()
-        if not self.dirLike:
+        if not self.isdir():
```

Comment:
forceDirectory=None adds a call to stat where previously no I/O was needed

Reply:
No. Prevoiusly it always required the stat on construction, now it requires the stat the first time it needs to know.

Reply:
Changed it back to look at `dirLike` as an optimization rather than something that changes behavior (because the profile showed that PosixPath has a long construction time).